Not my errors:

```
[ 2 of 18] Compiling Data.Machine.Plan ( src/Data/Machine/Plan.hs, dist/build/Data/Machine/Plan.o, dist/build/Data/Machine/Plan.dyn_o )

src/Data/Machine/Plan.hs:179:14: error:
Error:     • Couldn't match type: PlanT k1 o0 m0 a0
                     with: forall (m :: * -> *). PlanT k o m ()
      Expected: Maybe o -> Plan k o ()
        Actual: Maybe o -> PlanT k1 o0 m0 a0
    • In the expression: maybe stop yield
      In an equation for ‘maybeYield’: maybeYield = maybe stop yield
    • Relevant bindings include
        maybeYield :: Maybe o -> Plan k o ()
          (bound at src/Data/Machine/Plan.hs:179:1)
    |
179 | maybeYield = maybe stop yield
    |              ^^^^^^^^^^^^^^^^

src/Data/Machine/Plan.hs:179:25: error:
Error:     • Couldn't match type: forall (m :: * -> *). PlanT k0 o m ()
                     with: PlanT k1 o0 m0 a0
      Expected: o -> PlanT k1 o0 m0 a0
        Actual: o -> Plan k0 o ()
    • In the second argument of ‘maybe’, namely ‘yield’
      In the expression: maybe stop yield
      In an equation for ‘maybeYield’: maybeYield = maybe stop yield
    • Relevant bindings include
        maybeYield :: Maybe o -> Plan k o ()
          (bound at src/Data/Machine/Plan.hs:179:1)
    |
179 | maybeYield = maybe stop yield
    |                         ^^^^^

src/Data/Machine/Plan.hs:203:28: error:
Error:     • Couldn't match type: forall (m1 :: * -> *). PlanT k2 a m1 ()
                     with: PlanT k a m a1
      Expected: Maybe a -> PlanT k a m a1
        Actual: Maybe a -> Plan k2 a ()
    • In the second argument of ‘(>>=)’, namely ‘maybeYield’
Error: cabal: Failed to build machines-0.7 (which is required by
test:crem-spec from crem-0.1.0.0 and test:crem-doctests from crem-0.1.0.0).
See the build log above for details.

      In a stmt of a 'do' block: (lift f >>= maybeYield)
      In the expression:
        do (lift f >>= maybeYield)
           exhaust f
    • Relevant bindings include
        f :: m (Maybe a) (bound at src/Data/Machine/Plan.hs:203:9)
        exhaust :: m (Maybe a) -> PlanT k a m ()
          (bound at src/Data/Machine/Plan.hs:203:1)
    |
203 | exhaust f = do (lift f >>= maybeYield); exhaust f
    |                            ^^^^^^^^^^
Error: Process completed with exit code 1.
```

--- a/crem.cabal
+++ b/crem.cabal
@@ -55,6 +55,7 @@ library
   ghc-options: -Weverything -Wno-safe -Wno-unsafe -Wno-missing-safe-haskell-mode -Wno-implicit-prelude -Wno-missing-export-lists -Wno-missing-home-modules -Wno-missing-import-lists -Wno-all-missed-specialisations -Wno-prepositive-qualified-module
   build-depends:
       base >=4.15 && <4.19
+    , machines >=0.7.3 && <0.8
     , nothunks >=0.1 && <0.4
     , profunctors >=3.2 && <5.7
     , singletons-base >=3.0 && <3.3
@@ -434,6 +435,7 @@ test-suite crem-spec
     , crem
     , crem-examples
     , hspec >=2.7 && <2.12
+    , machines
     , profunctors
     , singletons-base
   default-language: Haskell2010

--- a/package.yaml
+++ b/package.yaml
@@ -126,6 +126,7 @@ library:
     - singletons-base >= 3.0 && < 3.3
     - text >= 1.2 && < 2.1
     - nothunks >= 0.1 && < 0.4
+    - machines >=0.7.3 && <0.8
   # Disable adding Paths_crem to other-modules, because it does not conform to our style guide.
   # https://github.com/sol/hpack#handling-of-paths_-modules
   when:
@@ -154,6 +155,7 @@ tests:
       - crem
       - crem-examples
       - hspec >= 2.7 && < 2.12
+      - machines
       - profunctors
       - singletons-base
     build-tools: hspec-discover:hspec-discover

--- a/spec/Crem/StateMachineSpec.hs
+++ b/spec/Crem/StateMachineSpec.hs
@@ -9,6 +9,9 @@ import "base" Data.Functor.Identity (Identity (..))
 import "base" Data.List (singleton)
 import "crem" Crem.StateMachine
 import "hspec" Test.Hspec (Expectation, Spec, describe, it, shouldBe)
+import "machines" Data.Machine (runT, source, (~>))
+import "machines" Data.Machine.Process (autoT)
+import "machines" Data.Machine.Process qualified
 import "profunctors" Data.Profunctor (rmap)
 import "singletons-base" Data.Singletons.Base.TH
 
@@ -116,3 +119,76 @@ spec =
 
         it "processes correctly multiple inputs" $ do
           runMultiple (Feedback echo plus1UpTo4) [1, 1] `shouldOutput` [1, 2, 3, 4, 5, 1, 2, 3, 4, 5]
+
+    describe "AutomatonM instance" $ do
+      describe "autoT converts StateMachineT to ProcessT" $ do
+        it "processes a single input through a stateless machine" $ do
+          let machine = stateless (+ 1) :: StateMachineT Identity Int Int
+          let process = autoT machine
+          let result = runIdentity $ runT $ source [5] ~> process
+          result `shouldBe` [6]
+
+        it "processes multiple inputs through a stateless machine" $ do
+          let machine = stateless (* 2) :: StateMachineT Identity Int Int
+          let process = autoT machine
+          let result = runIdentity $ runT $ source [1, 2, 3, 4] ~> process
+          result `shouldBe` [2, 4, 6, 8]
+
+        it "processes inputs through a stateful machine (switch)" $ do
+          let machine = Basic $ switchMachine SFalse :: StateMachineT Identity () ()
+          let process = autoT machine
+          let result = runIdentity $ runT $ source [(), (), ()] ~> process
+          result `shouldBe` [(), (), ()]
+
+        it "maintains state across inputs (lock door)" $ do
+          let machine = Basic $ lockDoorMachine SIsLockClosed :: StateMachineT Identity LockDoorCommand LockDoorEvent
+          let process = autoT machine :: Data.Machine.Process.ProcessT Identity LockDoorCommand LockDoorEvent
+          let result = runIdentity $ runT $ source [LockLock, LockUnlock, LockOpen] ~> process
+          result `shouldBe` [LockLocked, LockUnlocked, LockOpened]
+
+        it "works with sequential composition" $ do
+          let machine1 = stateless (+ 1) :: StateMachineT Identity Int Int
+          let machine2 = stateless (* 2) :: StateMachineT Identity Int Int
+          let composed = machine2 Control.Category.. machine1 -- Category composition: right-to-left
+          let process = autoT composed
+          let result = runIdentity $ runT $ source [1, 2, 3] ~> process
+          result `shouldBe` [4, 6, 8]
+
+        it "works with parallel composition" $ do
+          let machine1 = stateless (+ 1) :: StateMachineT Identity Int Int
+          let machine2 = stateless (* 2) :: StateMachineT Identity Int Int
+          let parallel = Parallel machine1 machine2
+          let process = autoT parallel
+          let result = runIdentity $ runT $ source [(1, 10), (2, 20), (3, 30)] ~> process
+          result `shouldBe` [(2, 20), (3, 40), (4, 60)]
+
+        it "works with alternative composition (left)" $ do
+          let machine1 = stateless (+ 1) :: StateMachineT Identity Int Int
+          let machine2 = stateless (* 2) :: StateMachineT Identity Int Int
+          let alt = Alternative machine1 machine2
+          let process = autoT alt
+          let result = runIdentity $ runT $ source [Left 1, Left 2, Left 3] ~> process
+          result `shouldBe` [Left 2, Left 3, Left 4]
+
+        it "works with alternative composition (right)" $ do
+          let machine1 = stateless (+ 1) :: StateMachineT Identity Int Int
+          let machine2 = stateless (* 2) :: StateMachineT Identity Int Int
+          let alt = Alternative machine1 machine2
+          let process = autoT alt
+          let result = runIdentity $ runT $ source [Right 1, Right 2, Right 3] ~> process
+          result `shouldBe` [Right 2, Right 4, Right 6]
+
+        it "works with mixed alternative composition" $ do
+          let machine1 = stateless (+ 1) :: StateMachineT Identity Int Int
+          let machine2 = stateless (* 2) :: StateMachineT Identity Int Int
+          let alt = Alternative machine1 machine2
+          let process = autoT alt
+          let result = runIdentity $ runT $ source [Left 1, Right 5, Left 2] ~> process
+          result `shouldBe` [Left 2, Right 10, Left 3]
+
+        it "handles boolean state machine correctly" $ do
+          let machine = booleanStateMachine SFalse :: StateMachineT Identity Int Int
+          let process = autoT machine :: Data.Machine.Process.ProcessT Identity Int Int
+          let result = runIdentity $ runT $ source [0, 1, 0] ~> process
+          -- Starting at SFalse: 0 (even) -> 1, stay SFalse; 1 (odd) -> 3, move to STrue; 0 (even) -> -1, stay STrue
+          result `shouldBe` [1, 3, -1]

--- a/src/Crem/StateMachine.hs
+++ b/src/Crem/StateMachine.hs
@@ -14,6 +14,8 @@ import "base" Control.Category (Category (..))
 import "base" Data.Bifunctor (Bifunctor (second), bimap)
 import "base" Data.Foldable (foldlM)
 import "base" Data.Kind (Type)
+import "machines" Data.Machine.MealyT qualified as Machines
+import "machines" Data.Machine.Process qualified as Machines
 import "nothunks" NoThunks.Class (NoThunks (..), allNoThunks)
 import "profunctors" Data.Profunctor (Choice (..), Profunctor (..), Strong (..))
 import "singletons-base" Data.Singletons (Demote, SingI, SingKind)
@@ -189,6 +191,15 @@ instance (Monad m) => Arrow (StateMachineT m) where
 instance (Monad m) => ArrowChoice (StateMachineT m) where
   left :: StateMachineT m a b -> StateMachineT m (Either a c) (Either b c)
   left = left'
+
+-- * AutomatonM
+
+instance Machines.AutomatonM StateMachineT where
+  autoT :: (Monad m) => StateMachineT m a b -> Machines.ProcessT m a b
+  autoT = Machines.autoT . go
+    where
+      go :: (Monad m) => StateMachineT m a b -> Machines.MealyT m a b
+      go sm = Machines.MealyT $ fmap (fmap go) . run sm
 
 -- * Run a state machine
 

